src/components/BasicLed.vue

```diff
@@ -211,6 +236,11 @@ export default {
   },
   
   computed: {
+
+    loading() {
+      return this.showprogress
```

Comment:
loading state seems complex
when you end up with multiple boolean state indicators, code breaks quickly because you have 2^n possible states

refactor this to use a single state var: `loadState`, and give it string values, such as: `initial`, `loading`, `error`, `done` and then use those to control what happens

Reply:
I think we should be good now - please, have a look.